tests/numeric/test_ScalingTransformer.py

```diff
@@ -1,398 +1,119 @@
-        """Test that the scaler attribute is set to the correct type given what is passed when initialising the transformer."""
-        x = ScalingTransformer(columns="b", scaler_type=scaler_type)
-
-        assert (
-            type(x.scaler) is scaler_actual
-        ), f"unexpected scaler_type set in init for {scaler_type}"
-
-    @pytest.mark.parametrize(
-        ("scaler_type", "scaler_type_str", "scaler_kwargs_value"),
-        [
-            ("min_max", "MinMaxScaler", {"feature_range": (0.5, 1.5)}),
-            ("max_abs", "MaxAbsScaler", {}),
-            (
-                "standard",
-                "StandardScaler",
-                {"with_mean": True, "with_std": True},
-            ),
-        ],
-    )
-    def test_scaler_initialised_with_scaler_kwargs(
-        self,
-        mocker,
-        scaler_type,
-        scaler_type_str,
-        scaler_kwargs_value,
-    ):
-        """Test that the scaler_type is initialised with the scaler_kwargs arguments."""
-        mocked = mocker.patch(
-            f"sklearn.preprocessing.{scaler_type_str}.__init__",
-            return_value=None,
-        )
-
-        ScalingTransformer(
-            columns="b",
-            scaler_type=scaler_type,
-            scaler_kwargs=scaler_kwargs_value,
-        )
-
-        assert mocked.call_count == 1, "unexpected number of calls to init"
-
-        call_args = mocked.call_args_list[0]
-        call_pos_args = call_args[0]
-        call_kwargs = call_args[1]
-
-        assert (
-            call_pos_args == ()
-        ), f"unexpected positional args in {scaler_type_str} init call"
-
-        assert (
-            call_kwargs == scaler_kwargs_value
-        ), f"unexpected kwargs in {scaler_type_str} init call"
-
-    def test_super_init_called(self, mocker):
-        """Test that super.__init__ called."""
-        expected_call_args = {
-            0: {
-                "args": (),
-                "kwargs": {"columns": ["a", "b"], "verbose": False},
-            },
-        }
+            ScalingTransformer(columns=["a"], scaler_type="invalid_scaler")
 
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "__init__",
-            expected_call_args,
-        ):
-            ScalingTransformer(
-                columns=["a", "b"],
-                scaler_type="standard",
-                verbose=False,
-            )
 
+class TestFit(BaseNumericTransformerFitTests):
+    """Tests for ScalingTransformer.fit()"""
 
-class TestCheckNumericColumns:
-    """Tests for the check_numeric_columns method."""
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "ScalingTransformer"
+        cls.scaler_type = "standard"
 
-    def test_exception_raised(self):
-        """Test an exception is raised if non numeric columns are passed in X."""
-        df = d.create_df_2()
+    def test_non_numeric_exception_raised(self):
```

Comment:
Q: This test is already inherited from BaseNumericTransformerFitTests so I don't think it is needed here.  Am I missing osmething that is not covered by the test in the base class? https://github.com/lvgig/tubular/blob/feature/refactor_ScalingTransformer_tests/tests/numeric/test_BaseNumericTransformer.py

Reply:
I checked the base class tests and saw that it doesn’t cover the specific error messages and checks needed for ScalingTransformer. The base test is more general and doesn’t handle the exact way ScalingTransformer raises errors with non-numeric data.

So, I kept the specific test in ScalingTransformer to make sure it’s working correctly.

Reply:
how so? as far as I can tell the handling should be the same.  fit calls super.fit which calls the same _check_numeric method so seems like something is going wrong if the error is returning something different here?

Reply:
had a bit more of a dig and it looks like we forgot to implement any tests for fit for LogTransformer so the inheritance of this test has not been tested before this.

Reply:
in this case the test doesn't work because the  `x = initialized_transformers[self.transformer_name]` takes the default initialised transformer for ScalingTransformer and that transformer has only columns = ['a'].  Since the df we want to check only has non numeric data in column b we are not throwing an error.  adding b to columns in conftest.minimal_attribute_dict['ScalingTransformer] seems to fix the issue without breaking other tests.

Reply:
Thank you for this. I have now resolved it and it's working fine now